src/routes/users_routes.js

```diff
@@ -796,10 +796,10 @@ router.post('/queuefeatured', ensureAuth, async (req, res) => {
 
   try {
     if (shouldUpdate) {
-      fq.replaceForUser(req.user.id, cube.id);
+      await fq.replaceForUser(req.user.id, cube.id);
```

Comment:
Without the awaits Errors thrown by the functions wouldn't be caught and so the frontend would say "success" even though the action failed